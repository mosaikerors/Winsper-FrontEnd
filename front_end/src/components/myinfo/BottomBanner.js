import React from "react";
import { Text, View, StyleSheet,Picker } from "react-native";
import { Button, Avatar, Divider, Card } from 'react-native-elements'
import { TouchableOpacity } from "react-native-gesture-handler";
import { StackActions, NavigationActions } from "react-navigation";

const styles = StyleSheet.create({
    border: {
        borderWidth: 1,
    },
    block: {
        width: 60,
    },
    text: {
        justifyContent: 'center',
        alignItems: 'center',
        height: '100%'
    },
    logout: {
        marginRight: 10,
    }
})

const logout = StackActions.reset({
    index: 0,
    actions: [
        NavigationActions.navigate({ routeName: 'LoggedOut' })
    ]
});

class BottomBanner extends React.Component {
    render() {
        return (
            <React.Fragment>
                <View style={{ flex: 1, flexDirection: 'column-reverse' }}>
                    <View style={{ height: 40, flexDirection: 'row' }}>
                        <View style={{ flexDirection: 'row' }}>
                            <View style={styles.block}>
                                <TouchableOpacity style={styles.text} onPress={() => this.props.navigation.navigate('Settings')}>
                                    <Text>设置</Text>
                                </TouchableOpacity>
                            </View>
                            <View style={styles.block}>
                                <TouchableOpacity style={styles.text}>
                                    <Text>夜间</Text>
                                </TouchableOpacity>
                            </View>
                            <View style={styles.block}>
                                <TouchableOpacity style={styles.text}>
                                    <Text>东19</Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                        <View style={{ flexDirection: 'row-reverse', flex: 1 }}>
                            <TouchableOpacity style={[styles.text, styles.logout]} onPress={() => this.props.navigation.dispatch(logout)}>
                                <Text>退出登录</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                    <Divider />
                </View>
            </React.Fragment>
        );
    }
}
export default BottomBanner;