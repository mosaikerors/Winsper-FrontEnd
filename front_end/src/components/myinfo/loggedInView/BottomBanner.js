import React from "react";
import { Text, View, StyleSheet,TouchableOpacity} from "react-native";
import { Divider } from 'react-native-elements';
import { StackActions, NavigationActions } from "react-navigation";

const styles = StyleSheet.create({
    border: {
        borderWidth: 1,
    },
    block: {
        width: 60,
    },
    text: {
        justifyContent: 'center',
        alignItems: 'center',
        height: '100%'
    },
    logout: {
        marginRight: 10,
    }
});

const logout = StackActions.reset({
    index: 0,
    actions: [
        NavigationActions.navigate({ routeName: 'LoggedOut' })
    ]
});

class BottomBanner extends React.Component {
    constructor(props){
        super(props);
    }

    render() {
        return (
            <React.Fragment>
                <View style={{ flex: 1, flexDirection: 'column-reverse' }}>
                    <View style={{ height: 40, flexDirection: 'row' }}>
                        <View style={{ flexDirection: 'row' }}>
                            <View style={styles.block}>
                                <TouchableOpacity style={styles.text} onPress={() => this.props.navigation.navigate('Settings')}>
                                    <Text>设置</Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                        <View style={{ flexDirection: 'row-reverse', flex: 1 }}>
                            <TouchableOpacity style={[styles.text, styles.logout]} onPress={() => this.props.navigation.dispatch(logout)}>
                                <Text>退出登录</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                    <Divider />
                </View>
            </React.Fragment>
        );
    }
}
export default BottomBanner;