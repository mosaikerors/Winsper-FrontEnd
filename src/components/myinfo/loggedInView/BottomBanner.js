import React from "react";
import { Text, View, StyleSheet, TouchableOpacity } from "react-native";
import { Divider, Overlay } from 'react-native-elements';
import { StackActions, NavigationActions } from "react-navigation";
import { connect } from "react-redux";
import theme from "../../../theme"
import FontAwesome from "react-native-vector-icons/FontAwesome";
import FontAwesome5 from "react-native-vector-icons/FontAwesome5";
import LinearGradient from 'react-native-linear-gradient';

const styles = StyleSheet.create({
    border: {
        borderWidth: 1,
    },
    block: {
        width: 60,
    },
    text: {
        justifyContent: 'center',
        alignItems: 'center',
        height: '100%'
    },
    logout: {
        marginRight: 10,
    }
});

const logout = StackActions.reset({
    index: 0,
    actions: [
        NavigationActions.navigate({ routeName: 'LoggedOut' })
    ]
});

const mapStateToProps = state => ({})
const mapDispatchToProps = dispatch => ({
    onLoggedOut: () =>
        dispatch({ type: 'LOGGED_OUT' })
})

class BottomBanner extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            showThemeSelector: false,
        }
        this.logout = this.logout.bind(this);
    }

    logout() {
        this.props.onLoggedOut();
        this.props.navigation.dispatch(logout)
    }

    render() {
        const { showThemeSelector } = this.state;
        return (
            <React.Fragment>
                <View style={{ flex: 1, flexDirection: 'column-reverse', backgroundColor: theme.palette.sky[0] }}>
                    <View style={{ height: 40, flexDirection: 'row', }}>
                        <View style={{ flexDirection: 'row' }}>
                            <View style={styles.block}>
                                <TouchableOpacity style={[styles.text, { borderWidth: 0 }]} onPress={() => this.props.navigation.navigate('Settings')}>
                                    <Text>设置</Text>
                                </TouchableOpacity>
                            </View>
                        </View>
                        <View style={{ flexDirection: 'row-reverse', flex: 1 }}>
                            <TouchableOpacity style={[styles.text, styles.logout]} onPress={this.logout}>
                                <Text>退出登录</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                    <Divider />
                </View>
            </React.Fragment>
        );
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(BottomBanner);