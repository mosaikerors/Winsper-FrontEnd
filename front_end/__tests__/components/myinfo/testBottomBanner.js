import React from 'react';
import renderer from 'react-test-renderer';
import BottomBanner from "../../../src/components/myinfo/BottomBanner";

test("renders bottom banner", ()=>{
    const tree = renderer.create(
        <BottomBanner />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});