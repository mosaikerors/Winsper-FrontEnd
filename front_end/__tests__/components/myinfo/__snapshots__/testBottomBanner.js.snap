// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bottom banner 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column-reverse",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "width": 60,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text>
            设置
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 60,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text>
            夜间
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 60,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text>
            东19
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row-reverse",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Text>
          退出登录
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#bcbbc1",
        "height": 0.5,
      }
    }
  />
</View>
`;
